cursive/src/backends/wasm.rs

```diff
@@ -0,0 +1,304 @@
+
+        Self::new(canvas)
+    }
+
+    fn to_cursive_event(event: web_sys::KeyboardEvent) -> Event {
+        match event.key_code() {
+            8 => Event::Key(Key::Backspace),
+            9 => Event::Key(Key::Tab),
+            13 => Event::Key(Key::Enter),
+            19 => Event::Key(Key::PauseBreak),
+            27 => Event::Key(Key::Esc),
+            33 => Event::Key(Key::PageUp),
+            34 => Event::Key(Key::PageDown),
+            35 => Event::Key(Key::End),
+            36 => Event::Key(Key::Home),
+            37 => Event::Key(Key::Left),
+            38 => Event::Key(Key::Up),
+            39 => Event::Key(Key::Right),
+            40 => Event::Key(Key::Down),
+            45 => Event::Key(Key::Ins),
+            46 => Event::Key(Key::Del),
+            101 => Event::Key(Key::NumpadCenter),
+            112 => Event::Key(Key::F1),
+            113 => Event::Key(Key::F2),
+            114 => Event::Key(Key::F3),
+            115 => Event::Key(Key::F4),
+            116 => Event::Key(Key::F5),
+            117 => Event::Key(Key::F6),
+            118 => Event::Key(Key::F7),
+            119 => Event::Key(Key::F8),
+            120 => Event::Key(Key::F9),
+            121 => Event::Key(Key::F10),
+            122 => Event::Key(Key::F11),
+            123 => Event::Key(Key::F12),
+            code => {
+                if let Some(c) = std::char::from_u32(code) {
+                    Event::Char(c)
+                } else { Event::Unknown(Vec::new()) }
+            }  
+        }
+    }
+}
+
+impl cursive_core::backend::Backend for Backend {
+    fn poll_event(self: &mut Backend) -> Option<Event> {
+        self.events.borrow_mut().pop_front()
+    }
+
+    fn set_title(self: &mut Backend, title: String) {
+        self.canvas.set_title(&title);
+    }
+
+    fn refresh(self: &mut Backend) {
+        // web_sys::console::time_with_label("refresh");
+        let data = self.buffer.borrow().clone();
+        paint(text_color_pairs_to_bytes(&data));
+        // web_sys::console::time_end_with_label("refresh");
+    }
+
+    fn has_colors(self: &Backend) -> bool {
+        true
+    }
+
+    fn screen_size(self: &Backend) -> Vec2 {
+        Vec2::new(self.width, self.height)
+    }
+
+    fn print_at(self: &Backend, pos: Vec2, text: &str) {
+        let color = (*self.color.borrow()).clone();
+        let mut buffer = self.buffer.borrow_mut();
+        for (i, c) in text.chars().enumerate() {
+            let x = pos.x + i;
+            buffer[self.width * pos.y + x] = TextColorPair::new(c, color.clone());
+        }
+    }
+
+    fn clear(self: &Backend, _color: cursive_core::theme::Color) {
+        let color = cursive_to_color(_color);
+        let pair = ColorPair {
+            front: color,
+            back: color,
+        };
+        let mut buffer = self.buffer.borrow_mut();
+        for i in 0..self.width * self.height {
+            buffer[i] = TextColorPair::new(' ', pair.clone());
+        }
```

Comment:
```suggestion
        self.buffer
            .borrow_mut()
            .fill(TextColorPair::new(' ', pair.clone()));
```
Using the provided `.fill()` is a better way, as it might get (easier) vectorized and generally benefits from optimizations in the standard library.